File: admin-backend/admin-shared/admin-shared-persistence/src/main/java/io/github/faustofan/admin/shared/persistence/model/TenantAware.java
package io.github.faustofan.admin.shared.persistence.model;

import org.babyfish.jimmer.sql.MappedSuperclass;

/**
 * 多租户实体，包含租户ID字段
 */
@MappedSuperclass
public interface TenantAware extends BaseEntity {
    long tenantId();
    // 审计字段：创建人和更新人
    long createdBy();
    long updatedBy();
}

File: src/main/kotlin/io/github/faustofan/admin/shared/domain/model/TenantAware.kt
package io.github.faustofan.admin.shared.domain.model

import org.babyfish.jimmer.sql.MappedSuperclass

/**
 * 租户隔离接口
 * 所有需要租户隔离的表都继承此接口
 */
@MappedSuperclass
interface TenantAware : BaseEntity {
    // 这里的 tenantId 通常不需要手动 set，
    // 而是通过 Jimmer 的 DraftInterceptor 自动注入，
    // 并在查询时通过 Global Filter 自动过滤。
    val tenantId: Long
}